src/main/rx/crsf.c

```diff
@@ -150,6 +182,25 @@ typedef struct crsfPayloadLinkstatistics_s {
     int8_t downlink_SNR;
 } crsfLinkStatistics_t;
 
+#if defined(USE_CRSF_V3)
+typedef struct crsfPayloadLinkstatisticsRx_s {
+    uint8_t downlink_RSSI_1;
+    uint8_t downlink_RSSI_1_percentage;
+    uint8_t downlink_Link_quality;
+    int8_t downlink_SNR;
+    uint8_t uplink_power;
+} crsfLinkStatisticsRx_t;
+
+typedef struct crsfPayloadLinkstatisticsTx_s {
+    uint8_t uplink_RSSI;
+    uint8_t uplink_RSSI_percentage;
+    uint8_t uplink_Link_quality;
+    int8_t uplink_SNR;
+    uint8_t downlink_power;
```

Comment:
`downlink_power` is not used. Please remove.

Reply:
done.

Reply:
`CRSF_FRAME_LINK_STATISTICS_TX_PAYLOAD_SIZE` is wrong again :D 
why don't you just use `sizeof(crsfLinkStatisticsTx_t)` ?

Reply:
and `uplink_FPS` is useless as well... should be removed.

Reply:
the size is indeed 6. we removed "downlink_power" because betaflight is not using it. but it's definitely used on some targets, e.g. the radios. 
BTW do u want to remove all the unused entries?

Reply:
Remove all unnecessary data to minimize transfer overhead. No need to keep radio/handset specific stuff for FC just for fun, imo. 
Purpose of this PR is to optimize communication with Betaflight firmware so let's do it :)

Reply:
Please, don't remove `uplink_TX_Power`. It's used in #10582.

Reply:
> Remove all unnecessary data to minimize transfer overhead. No need to keep radio/handset specific stuff for FC just for fun, imo.
> Purpose of this PR is to optimize communication with Betaflight firmware so let's do it :)

Oh i think there is a misunderstanding here. let me try to clarify this. :smile: 
this telemetry frame will also be used on other devices like the VTX & radios. some of the telemetry entries are not used in betaflight but they are actually used on some other targets. in order to keep the consistency, those "unused" entries will also be transmitted.

Reply:
> `CRSF_FRAME_LINK_STATISTICS_TX_PAYLOAD_SIZE` is wrong again :D
> why don't you just use `sizeof(crsfLinkStatisticsTx_t)` ?

I agree, `sizeof(...)` is much more robust, especially when the struct is `__attribute__((packed))`
Currently the struct is aligned also when packed, however, when e.g. adding an uint32_t into the struct,
then counting by hand the sizes is tricky, see e.g.

```

#include <stdint.h>
#include <stdio.h>

typedef struct packed_crsfPayloadLinkstatisticsRx_s {
	uint8_t downlink_RSSI_1;
	uint8_t downlink_RSSI_1_percentage;
	uint8_t downlink_Link_quality;
	int8_t downlink_SNR;
	uint8_t uplink_power;
#ifdef FUTURE
	uint32_t future_extension;
#endif
} __attribute__((packed)) packed_crsfLinkStatisticsRx_t;

typedef struct crsfPayloadLinkstatisticsRx_s {
	uint8_t downlink_RSSI_1;
	uint8_t downlink_RSSI_1_percentage;
	uint8_t downlink_Link_quality;
	int8_t downlink_SNR;
	uint8_t uplink_power;
#ifdef FUTURE
	uint32_t future_extension;
#endif
} crsfLinkStatisticsRx_t;

int main(void) {
  printf("not packed %lu\n", sizeof(crsfLinkStatisticsRx_t));
  printf("packed %lu\n", sizeof(packed_crsfLinkStatisticsRx_t));

  return 0;
}
```

```
[tstibor@x230 ~]$ gcc -m32 crsf_sizeof_example.c -o crsf_sizeof_example && ./crsf_sizeof_example    
not packed 5
packed 5
[tstibor@x230 ~]$ gcc -m32 -DFUTURE crsf_sizeof_example.c -o crsf_sizeof_example && ./crsf_sizeof_example
not packed 12
packed 9
```

Reply:
@tstibor yeah, all struct should be packet here. Anyway that was just a suggestion to avoid misaligned sizes.

@tbs-fpv please add a comment then to avoid confusion. This is open source project so info is needed :)

Reply:
@tstibor thanks for your suggestions. 
this frame is also sent to other devices e.g. OSDs & radios. so the frame structure is somehow fixed. even betaflight is not taking the value of "downlink_power", we cannot remove it in the actual transmission or change the memory structure.

@cruwaller should we add comments to all unused members as remarks? what do u think?

Reply:
@tbs-fpv I think it would be good to keep those in the struct then. Just make a comment that belongs to transmitted struct but not used :)

btw, I tested this and works as expected. Negotiated baudrate configuration is ok as well.

Reply:
@cruwaller comments have been added. thanks for your time for testing it. :smiley: